common/models/src/utils.rs

```diff
@@ -99,6 +99,9 @@ impl SeqIdGenerator {
     }
 }
 
-pub fn build_address(address: String, port: u16) -> String {
-    format!("{}:{}", address, port)
+pub fn build_address(address: String, port: Option<u16>) -> String {
```

Comment:
For parameter `address` I think we don't need to take an owned String. And, build an empty string as the address is dangerous. Try this:

```rust
pub fn build_address(address: &str, port: u16) -> String {
    format!("{address}:{port}")
}
```